babble/src/main/java/io/mosaicnetworks/babble/servicediscovery/webrtc/WebRTCService.java

```diff
@@ -48,63 +50,95 @@
 import io.mosaicnetworks.babble.servicediscovery.ServiceDiscoveryListener;
+import io.mosaicnetworks.babble.utils.HttpsTrustManager;
+
 
 /**
  * Service class to handle WebRTC
  */
 public class WebRTCService implements ServiceAdvertiser {
 
-    public static final String DISCOVER_SERVER_HOST = "disco.babble.io";
-    public static final String RELAY_SEVER_ADDRESS = "disco.babble.io:2443";
+    private static final String TAG = "WebRTCService";
+    // XXX localhost values
+    // TODO these should not be hardcoded
+    public static final String DISCOVER_SERVER_HOST = "disco-staging.babble.io";
     public static final int DISCOVER_SERVER_PORT = 1443;
+    public static final String RELAY_SEVER_ADDRESS = "disco-staging.babble.io:2443";
 
     private static final String DISCOVER_END_POINT = "groups";
     private static final String REGISTER_END_POINT = "group";
 
+    private static String mGroupsURL;
+
+    // XXX Unsafe.
+    // TODO this should not be hardcoded
+    private static final boolean SKIP_VERIFY = true;
+
     private static RequestQueue sQueue;
     private ServiceDiscoveryListener mServiceDiscoveryListener;
 
     private static WebRTCService INSTANCE;
-    private List<ResolvedGroup> mResolvedGroups;
 
+    private List<ResolvedGroup> mResolvedGroups;
+    private HashMap<String, Boolean> mResolvedGroupsIndex;
     /**
      * Factory for the {@link WebRTCService}
      *
      * @return a WebRTCService service
      */
     public static WebRTCService getInstance(Context context) {
         if (INSTANCE == null) {
-            INSTANCE = new WebRTCService();
-            sQueue = Volley.newRequestQueue(context.getApplicationContext());
+            INSTANCE = new WebRTCService(context);
         }
         return INSTANCE;
     }
 
-    private WebRTCService() {}
+    private WebRTCService(Context context) {
+
+        sQueue = Volley.newRequestQueue(context.getApplicationContext());
+
+        // calculate groups URI once and for all
+        mGroupsURL = String.format(
+                "https://%s:%d/%s?app-id=%s",
+                DISCOVER_SERVER_HOST,
+                DISCOVER_SERVER_PORT,
+                DISCOVER_END_POINT,
+                context.getApplicationContext().getPackageName()
+        );
+
+        // disable TLS verification if skip-verify is set
+        if (SKIP_VERIFY) {
+            HttpsTrustManager.allowAllSSL();
+        }
+    }
 
     public void registerServiceDiscoveryListener(ServiceDiscoveryListener listener){
         mServiceDiscoveryListener = listener;
     }
 
     public void setResolvedGroups(List<ResolvedGroup> resolvedGroups) {
         mResolvedGroups = resolvedGroups;
+        mResolvedGroupsIndex = new HashMap<String, Boolean>();
+    }
+
+    public void addGroup(ResolvedGroup group) {
+        mResolvedGroups.add(group);
+        mResolvedGroupsIndex.put(group.getGroupUid(), true);
+    }
+
+    public void removeGroup(ResolvedGroup group) {
```

Comment:
Can removeGroup and addGroup be made private?

Reply:
done